### ISSUE TITLE
vue/require-v-for-key triggers on <template> tags

### ISSUE THREAD

**Issue by braddialpad** [vuejs/eslint-plugin-vue#2305]
**What rule do you want to change?**
vue/require-v-for-key

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Change of existing default behaviour

**Please provide some example code that this change will affect:**

```vue
<template v-for="i in 5">
  <!-- template tags do not require a key, however the linter reports it as an error. -->
  <div />
</template>
```

**What does the rule currently do for this code?**
Triggers lint error "Elements in iteration expect to have 'v-bind:key' directives."

**What will the rule do after it's changed?**
Will not trigger this rule in the case where the v-for is on a `<template>` tag.

**Additional context**

If you put a v-for on a `<template>` tag v-for does NOT need to have a key. This is shown directly in the documentation examples for vue 2 and vue 3, and through testing it appears to work as expected if you ignore the linter error.

https://vuejs.org/guide/essentials/list.html#v-for-on-template
https://v2.vuejs.org/v2/guide/list.html#v-for-on-a-lt-template-gt

**Comment by FloEdelmann** [comment]
That is wrong. See the section a bit further down in the docs, there is an example of `<template>` with `key`: https://vuejs.org/guide/essentials/list.html#maintaining-state-with-key

If you don't like it, you can disable the rule.
